lib/_http_agent.js

```diff
@@ -319,6 +326,22 @@ function installListeners(agent, s, options) {
   }
   s.on('close', onClose);
 
+  function onTimeout() {
+    debug('CLIENT socket onTimeout');
+
+    const sockets = agent.freeSockets;
+    for (const name of ObjectKeys(sockets)) {
```

Comment:
I think `Object.values` will be a better fit here.
Or we can just use a `for..in` loop with the same behavior and without creating unnecessary arrays for iteration (since we are only interested in the first-found-socket).

Reply:
See comment below